lldb/source/Plugins/Language/Swift/SwiftFormatters.cpp

```diff
@@ -950,6 +954,135 @@ class TaskSyntheticFrontEnd : public SyntheticChildrenFrontEnd {
   ValueObjectSP m_child_tasks_sp;
   ValueObjectSP m_is_running_sp;
 };
+
+class TaskGroupSyntheticFrontEnd : public SyntheticChildrenFrontEnd {
+public:
+  TaskGroupSyntheticFrontEnd(lldb::ValueObjectSP valobj_sp)
+      : SyntheticChildrenFrontEnd(*valobj_sp.get()) {}
+
+  llvm::Expected<uint32_t> CalculateNumChildren() override {
+    return m_group_tasks.size();
+  }
+
+  bool MightHaveChildren() override { return true; }
+
+  lldb::ValueObjectSP GetChildAtIndex(uint32_t idx) override {
+    if (!m_ts || idx >= m_group_tasks.size())
+      return {};
+
+    if (auto valobj_sp = m_children[idx])
+      return valobj_sp;
+
+    // TypeMangling for "Swift.UnsafeCurrentTask"
+    CompilerType task_type =
+        m_ts->GetTypeFromMangledTypename(ConstString("$sSctD"));
+
+    addr_t task_addr = m_group_tasks[idx];
+    auto task_sp = ValueObject::CreateValueObjectFromAddress(
+        "current_task", task_addr, m_backend.GetExecutionContextRef(),
+        task_type, false);
+    if (auto synthetic_sp = task_sp->GetSyntheticValue())
+      task_sp = synthetic_sp;
+
+    task_sp->SetName(ConstString(("[" + Twine(idx) + "]").str()));
```

Comment:
Why the detour through ConstString?

Reply:
`SetName` takes a `ConstString`.

Reply:
Sorry! I misread the nesting of the parentheses and thought the .str() was in the outmost layer.